File: JS/Sorting/selectionSort.js
function selectionSort(arr) {
  for (let i = 0; i < arr.length; i++) {
    let min = i;
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[j] < arr[min]) {
        min = j;
      }
    }
    if (i !== min) {
      [ arr[i], arr[min] ] = [ arr[min], arr[i] ];
    }
  }
  return arr;
}

console.log(selectionSort([1, 4, 5, 6, 7, 4, 3, 2, 5, 6, 7, 8]));

File: CSharp/Sorting/SelectionSort.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CSharp.Sorting
{
    public class SelectionSort
    {
        public static int[] Sort(int[] arr)
        {
            for (var i = 0; i < arr.Length; i++)
            {
                var min = i;

                for (var j = i + 1; j < arr.Length; j++)
                {
                    if (arr[j] < arr[min])
                    {
                        min = j;
                    }
                }

                if (min != i)
                {
                    var temp = arr[i];
                    arr[i] = arr[min];
                    arr[min] = temp;
                }
            }

            return arr;
        }
    }
}
